akka-http-caching/src/main/java/akka/http/caching/javadsl/Cache.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.http.caching.javadsl;
+
+import akka.annotation.ApiMayChange;
+import akka.annotation.DoNotInherit;
+import akka.japi.Creator;
+import akka.japi.Procedure;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@ApiMayChange
+@DoNotInherit
+public interface Cache<K, V> {
+  /**
+   * Returns either the cached CompletionStage for the given key or evaluates the given value generating
+   * function producing a `CompletionStage<V>`.
+   */
+  CompletionStage<V> getFuture(K key, Creator<CompletionStage<V>> genValue);
+
+  /**
+   * Returns either the cached {@code CompletionStage} for the key, or evaluates the given function which
+   * should lead to eventual completion of the completable future.
+   */
+  CompletionStage<V> getOrFulfil(K key, Procedure<CompletableFuture<V>> f);
```

Comment:
Rename to `getOrComplete` to better match wording used regarding Java/Scala futures.

Reply:
Yeah good point